cvmfs/statistics_database.cc

```diff
@@ -331,45 +375,123 @@ StatisticsDatabase::~StatisticsDatabase() {
 }
 
 
-int StatisticsDatabase::StoreStatistics(const perf::Statistics *statistics,
-                                        const std::string &start_time,
-                                        const std::string &finish_time,
-                                        const std::string &command_name,
-                                        const std::string &repo_name) {
-  std::string insert_statement;
-  if (command_name == "ingest" || command_name == "sync") {
-    insert_statement = PrepareStatementIntoPublish(statistics, start_time,
-                                                               finish_time);
-  } else if (command_name == "gc") {
-    insert_statement = PrepareStatementIntoGc(statistics, start_time,
-                                              finish_time, repo_name);
+StatisticsDatabase *StatisticsDatabase::OpenStandardDB(
+  const std::string repo_name)
+{
+  StatisticsDatabase *db;
+  std::string db_file_path = GetDBPath(repo_name);
+  if (FileExists(db_file_path)) {
+    db = StatisticsDatabase::Open(db_file_path, kOpenReadWrite);
+    if (db == NULL) {
+      LogCvmfs(kLogCvmfs, kLogSyslogErr,
+                "Couldn't create StatisticsDatabase object!");
+      return NULL;
+    }
   } else {
-    return -5;
+    db = StatisticsDatabase::Create(db_file_path);
+    if (db == NULL) {
+      LogCvmfs(kLogCvmfs, kLogSyslogErr,
+               "Couldn't create StatisticsDatabase object!");
+      return NULL;
+    // insert repo_name into properties table
+    } else if (!db->SetProperty("repo_name", repo_name)) {
+      LogCvmfs(kLogCvmfs, kLogSyslogErr,
+               "Couldn't insert repo_name into properties table!");
+      return NULL;
+    }
   }
+  db->repo_name_ = repo_name;
+  return db;
+}
+
+
+bool StatisticsDatabase::StorePublishStatistics(
+  const perf::Statistics *statistics,
+  const std::string &start_time,
+  const bool success)
+{
+  std::string finish_time = GetGMTimestamp();
+  std::string statement = PrepareStatementIntoPublish(statistics, start_time,
+                                                      finish_time, success);
+  return StoreEntry(statement);
+}
 
+
+bool StatisticsDatabase::StoreGCStatistics(
+  const perf::Statistics *statistics,
+  const std::string &start_time,
+  const bool success)
+{
+  std::string finish_time = GetGMTimestamp();
+  std::string statement =
+    PrepareStatementIntoGc(statistics, start_time, finish_time,
+                           repo_name_, success);
```

Comment:
Perhaps this can be made symmetric to the `StorePublishStatistics`, i.e. without the repo_name parameter